XenonCode.hpp

```diff
@@ -5792,21 +5792,25 @@ const int VERSION_PATCH = 0;
 		std::unordered_map<uint32_t/*24 least significant bits only*/, std::string> Device::deviceFunctionNamesById {};
 		std::unordered_map<uint32_t/*24 least significant bits only*/, DeviceFunction> Device::deviceFunctionsById {};
 		std::unordered_map<uint8_t, std::vector<std::string>> Device::deviceFunctionsList {};
+		std::unordered_map<uint32_t, uint32_t> currentLineByAddr;
+		std::unordered_map<uint32_t, std::string> currentFileByAddr;
```

Comment:
These must be reset when the virtual computer is rebooted, otherwise it may give false information and corrupt the program if a file was modified. This would cause an important bug. 
Maybe clear them at the end of `Bootup()`?

Also, this information (file and line) is only used for error reporting so it's a waste anyways to compute it even once, when there are no errors in the program. 
Maybe a more appropriate although more involved "fix" would be a lazy loading upon error?
Also there is currently a bug with error reporting where sometimes the wrong line number is used.. that's a more pressing issue to fix. 
But your current optimization seems fair for now and adds a good value, as long as it doesn't cause additional bugs (as mentioned above).

Reply:
This resets on bootup [0fe12b6](https://github.com/batcholi/XenonCode/pull/6/commits/0fe12b6aa076af6dc19046606f0b9b67187b0900)

Yeah lazy loading on error definitely makes sense but that would've been a much larger PR